spritesheet-manager.js

```diff
@@ -29,13 +34,43 @@ class SpritesheetManager {
     }
     return spritesheet;
   }
+
   async getSpriteSheetForAppUrlAsync(appUrl, opts) {
-    let spritesheet = this.spritesheetCache.get(appUrl);
-    if (!spritesheet) {
-      spritesheet = await this.getSpriteSheetForAppUrlInternal([appUrl, opts]);
-      this.spritesheetCache.set(appUrl, spritesheet);
+    const spritesheet = this.spritesheetCache.get(appUrl);
+
+    if (spritesheet) return spritesheet
+
+    return new Promise((resolve) => {
+      this.waitQueue.push({ url: appUrl, opts, resolve })
+      if (!this.isLoaderRunning) this.loadSpriteSheetFromQueue()
+    })
+  }
+
+  async loadSpriteSheetFromQueue() {
+    if (this.currentlyLoading >= this.maxConcurrentLoading) return;

```

Comment:
For this counted dispatch technique, my philosophy is you have to decide who is doing the counting and the dispatching check. Is it the caller or the method?

Here the check seems to be split across the caller and the method, which is not good. Should either do all of the checking here or outside (possibly in its own method).

Reply:
Talking about this: https://github.com/webaverse/app/pull/3686/files#diff-7c16ad233a6efe359517c683cb8107f3f2e01df21e1b3084350926627486656aR64

Reply:
Very helpful insight. I updated the logic accordingly and now there is only one method which handles checks.